### ISSUE TITLE
TypeError instead of NotAllowedError in navigator.clipboard.write(image/png DOMString)

### ISSUE THREAD

**Issue by karlcow** [web-platform-tests/wpt#49301]
https://github.com/web-platform-tests/wpt/blob/92f1c298bc1d3c7915144d136c9b13c9be8e80e6/clipboard-apis/async-navigator-clipboard-basics.https.html#L156-L160

The current specificiation for write(data)
https://w3c.github.io/clipboard-apis/#dom-clipboard-write

In Steps 4, it says:
> 4. For each clipboardItem in dataList:

and then in 4.1.2.*

> 2. If representationDataPromise was rejected, then:
>     1.[Reject](https://webidl.spec.whatwg.org/#reject) p with ["NotAllowedError"](https://webidl.spec.whatwg.org/#notallowederror) [DOMException](https://webidl.spec.whatwg.org/#idl-DOMException) in realm.
>     2. Abort these steps. 

Here the test is assessing a TypeError instead of NotAllowedError. 
Is there another step I'm missing? 

@saschanaz @whsieh

**Comment by saschanaz** [comment]
`representationDataPromise` is coming from:

> 1. Let representationDataPromise be the representation’s [data](https://w3c.github.io/clipboard-apis/#representation-data).

And that `data` is from the [ClipboardItem constructor](https://w3c.github.io/clipboard-apis/#dom-clipboarditem-clipboarditem)

> 2. Set representation’s [data](https://w3c.github.io/clipboard-apis/#representation-data) to value.

There's no defined step to throw any error for the string-typed image data, per what I see. @evilpie, any idea?

**Comment by evilpie** [comment]
Yeah. Per spec this should probably not be an error. Maybe that should change though. Reinterpreting DOMString as bytes for an image seems weird.  I think when I implemented this, Chrome didn't support strings at all, so we with a more minimal approach.

**Comment by karlcow** [comment]
@evilpie @saschanaz So this looks like a spec discussion, more than a test discussion. 
What about the test change itself. Should we go ahead and accept the modifications I did which is following the spec.
Or should I modify something on my test change?

**Comment by karlcow** [comment]
> I see no point to change the test when the spec doesn't say much. We can split the test into tentative.

hmm indeed that could be a solution. 
Oh and the spec markup is confusing here. 

https://w3c.github.io/clipboard-apis/#dom-clipboarditem-clipboarditem

> 6. For each (key, value) in items:

Then it enumerate 1 to 6. 
and then greenbox
and then again 1 to 6, 

should it be 7 to 12. or is it a different sequence. 

Anyway thanks a lot to both of you. I will adjust the PR.
